packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DatePickerInput.tsx

```diff
@@ -100,55 +100,47 @@ const DatePickerInput = React.forwardRef((props, forwardedRef) => {
   );
 
   return (
-    <Popover
-      content={
-        <DatePicker
-          date={date}
-          onChange={onDateSelected}
-          setFocus
-          isDateDisabled={isDateDisabled}
-          localizedNames={localizedNames}
-        />
-      }
-      placement='bottom'
-      visible={isVisible}
-      onClickOutside={(_, e) => {
-        if (!buttonRef.current?.contains(e.target as Node)) {
-          close();
-        }
-      }}
-      appendTo='parent'
-    >
-      <InputGrid labelPlacement='inline' {...wrapperProps}>
-        <Label
-          as='label'
+    <InputGrid labelPlacement='inline' {...wrapperProps}>
+      <Label
+        as='label'
+        required={required}
+        disabled={disabled}
+        htmlFor={id}
+        {...labelProps}
+      >
+        {label}
+      </Label>
+      <InputWithDecorations {...inputWrapperProps}>
+        <InputWithDecorations.Input
+          id={id}
+          value={inputValue}
+          onChange={onInputChange}
           required={required}
           disabled={disabled}
-          htmlFor={id}
-          {...labelProps}
+          ref={forwardedRef}
+          {...rest}
+        />
+        <Popover
+          content={
+            <DatePicker
+              date={date}
+              onChange={onDateSelected}
+              setFocus
+              isDateDisabled={isDateDisabled}
+              localizedNames={localizedNames}
+            />
+          }
+          placement='bottom-end'
+          visible={isVisible}
+          onVisibleChange={setIsVisible}
+          closeOnOutsideClick
         >
```

Comment:
Any reason to move `<Popover>` inside instead of letting it be the top most component? Something similar to https://github.com/iTwin/iTwinUI/pull/1506#discussion_r1321671312?

Reply:
same response here as well. `referenceProps` gets applied on the cloned children and it includes aria attributes which should go on the correct element